libsolidity/analysis/TypeChecker.cpp

```diff
@@ -4033,6 +4033,10 @@ void TypeChecker::endVisit(Literal const& _literal)
 					m_errorReporter.typeError(
 						2505_error,
 						_literal.location(),
+						SecondarySourceLocation().append(
+							"Function was defined here:",
+							dynamic_cast<FunctionDefinition const*>(&suffixFunctionType->declaration())->location()
```

Comment:
Might be better to have it point just at the parameter list rather than the whole function.

Same below.